relay-server/src/services/buffer/envelope_store/sqlite.rs

```diff
@@ -32,12 +32,48 @@ use tokio::time::sleep;
 const ZSTD_MAGIC_WORD: &[u8] = &[40, 181, 47, 253];
 
 /// Struct that contains all the fields of an [`Envelope`] that are mapped to the database columns.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct DatabaseEnvelope {
     received_at: i64,
     own_key: ProjectKey,
     sampling_key: ProjectKey,
-    encoded_envelope: Vec<u8>,
+    encoded_envelope: Box<[u8]>,
+}
+
+#[derive(Clone, Debug)]
+pub struct DatabaseBatch {
+    received_at: i64,
+    own_key: ProjectKey,
+    sampling_key: ProjectKey,
+    envelopes: Vec<DatabaseEnvelope>,
+}
+
+impl DatabaseBatch {
+    pub fn len(&self) -> usize {
+        self.envelopes.len()
+    }
+}
+
+impl TryFrom<Vec<DatabaseEnvelope>> for DatabaseBatch {
+    type Error = ();
+
+    fn try_from(envelopes: Vec<DatabaseEnvelope>) -> Result<Self, Self::Error> {
+        let Some(newest) = envelopes.last() else {
```

Comment:
Is it safe to assume that the last is the most recent?

Reply:
It might not be, but it is consistent with the behavior of the priority queue wrt stacks: the priority of the stack is determined by the top element.

I renamed the variable to be less misleading.